### DISCUSSION TITLE
How to save multiple files without subsequent prompts?

### DISCUSSION THREAD

**Discussion by Simeon231** [CommunityToolkit/Maui#1669] [Q&A]
Hello,

My goal is to download many files on an Android device with just one prompt, without any prompts would be even better. I use FileSaver.Default and after invoking the SaveAsync method a prompt is shown (check the image) and for each subsequent method call the prompt is shown again. I tried:
- Passing an initial path, the same result.
- FolderPicker.Default, the UX is even worse because the user has to pick a folder twice for each file.
- On Android emulator (the photo is done with the emulator).
- On my real phone which is with Android 11.

I use .NET 7 (Can upgrade to 8 if it is needed).

Is there a way to avoid these prompts?

![image](https://github.com/CommunityToolkit/Maui/assets/50376134/df14097c-4c41-430e-b0fe-3820a0c92121)

**Comment by ghost** [comment]
Can you ask in the Popup for permission/prompt and then download files and save them in the code, without prompting the user?
I guess you would need WRITE_EXTERNAL_STORAGE permission. [Blog topic on basic File saving](https://askxammy.com/saving-files-with-net-maui-community-toolkit/). You can also read more [detailed discussion here](https://github.com/CommunityToolkit/Maui/discussions/1046) - related to this topic.
